src/Com/NoSyn/Environment/ProgramEnvironment.hs

```diff
@@ -66,11 +69,14 @@ lookupDType' noSynType aliasEnvironment
 lookupAtomicNoSynType::Ident->AliasEnvironment->CompilerStatus Ident
 lookupAtomicNoSynType noSynType aliasEnvironment
     | noSynType `OrderMap.member` aliasEnvironment = do
-      nextAlias <- compilerStatusFromMaybe ("COMPILER ERROR: imported library incorrect") (OrderMap.lookup noSynType aliasEnvironment)
-      lookupAtomicNoSynType' noSynType nextAlias aliasEnvironment
+        nextAlias <- compilerStatusFromMaybe ("COMPILER ERROR: imported library incorrect") (OrderMap.lookup noSynType aliasEnvironment)
+        lookupAtomicNoSynType' noSynType nextAlias aliasEnvironment
     | lastChar noSynType == "*" && (dropLastChar noSynType) `OrderMap.member` aliasEnvironment = do
-      nextAlias <- compilerStatusFromMaybe ("COMPILER ERROR: imported library incorrect") (OrderMap.lookup (dropLastChar noSynType) aliasEnvironment)
-      lookupAtomicNoSynType' (dropLastChar noSynType) nextAlias aliasEnvironment >>= \x -> return (x++"PTR")
+        nextAlias <- compilerStatusFromMaybe ("COMPILER ERROR: imported library incorrect") (OrderMap.lookup (dropLastChar noSynType) aliasEnvironment)
```

Comment:
when the type system gets changed to a tree based system, a more generic approach to checking the acceptability of type modifiers can be used in this function #21